src/config_stash/config.py

```diff
@@ -0,0 +1,104 @@
+import os
+from typing import Any
+from typing import List
+
+from src.config_stash.loaders import EnvLoader
+from src.config_stash.loaders import MultipleEnvLoader
+from src.config_stash.loaders import PrefixedEnvLoader
+from src.config_stash.loaders import VaultLoader
+from src.config_stash.loaders import YamlLoader
+
+
+class Config(dict):
+    """
+    A configuration class that extends the built-in Python dictionary.
+    It provides methods to load configuration values from various sources
+    such as environment variables, YAML files, and Vault secrets.
+    """
+
+    def __init__(self, vault_fetcher: Any = None, *args, **kwargs):
+        """
+        Initialize a new Config object.
+
+        Args:
+            vault_fetcher (Any, optional): An object used to fetch secrets from Vault. Defaults to None.
+            *args: Positional arguments to initialize the parent dictionary.
+            **kwargs: Keyword arguments to initialize the parent dictionary.
+        """
+        super().__init__(*args, **kwargs)
+        self.vault_fetcher = vault_fetcher
+
+    def load_many_keys_from_env(self, keys: List[str], loader: Any = MultipleEnvLoader()):
+        """
+        Load multiple keys from environment variables using the specified loader.
+
+        Args:
+            keys (List[str]): List of environment variable keys to load.
+            loader (Any, optional): Loader object to load environment variables. Defaults to MultipleEnvLoader().
+        """
+        values = loader.load(keys)
+        self.update(values)
+
+    def load_from_env(self, key: str, custom_key_name: str = None, loader: Any = EnvLoader()):
+        """
+        Load a key from an environment variable using the specified loader.
+
+        Args:
+            key (str): Environment variable key to load.
+            custom_key_name (str, optional): Optional custom key name for the loaded value. Defaults to None.
+            loader (Any, optional): Loader object to load the environment variable. Defaults to EnvLoader().
+        """
+        value = loader.load(key)
+        if not custom_key_name:
+            self[key] = value
+        else:
+            self[custom_key_name] = value
+
+    def load_prefixed_env_vars(self, allowed_prefixes: List[str] = None, loader: Any = PrefixedEnvLoader()):
+        """
+        Load environment variables with specified prefixes using the specified loader.
+
+        Args:
+            allowed_prefixes (List[str], optional): List of allowed prefixes for environment variables.
+            Defaults to None.
+            loader (Any, optional): Loader object to load environment variables.
+            Defaults to PrefixedEnvLoader().
+        """
+        values = loader.load(allowed_prefixes)
+        for key, value in values.items():
+            if key not in self:
+                self[key] = value
+
+    def load_from_yaml_file(self, filepath: str, loader: Any = YamlLoader()):
```

Comment:
When loading an empty yaml file, getting below log trace
```python
>>> config = Config()
>>> config.load_from_yaml_file("test.yml")
```


```bash
  File "/Users/gnagasai/go/src/github.com/config-stash/src/config_stash/loaders.py", line 132, in load
    self._load_yaml_data(data, vault_fetcher=vault_fetcher)
  File "/Users/gnagasai/go/src/github.com/config-stash/src/config_stash/loaders.py", line 139, in _load_yaml_data
    for key, value in data.items():
```

please add an appropriate error when loaded file is empty